src/java/org/apache/cassandra/tcm/ClusterMetadataService.java

```diff
@@ -373,6 +373,20 @@ public void reconfigureCMS(ReplicationParams replicationParams)
         InProgressSequences.finishInProgressSequences(ReconfigureCMS.SequenceKey.instance);
     }
 
+    public void maybeReconfigureCMS(ClusterMetadata metadata)
+    {
+        try
+        {
+            if (PrepareCMSReconfiguration.needsReconfiguration(metadata))
+                reconfigureCMS(ReplicationParams.meta(metadata));
+        }
+        catch (Throwable t)
+        {
+            JVMStabilityInspector.inspectThrowable(t);
+            logger.error("Could not reconfigure CMS, operator should run `nodetool cms reconfigure` to make sure CMS placement is correct", t);
```

Comment:
if we do make a change to submit the reconfig unconditionally, it would be good to still log something like this in the event that the submission fails, maybe at `WARN`?